tests/temp_tests/test_settings/test_mpiLauncher.py

```diff
@@ -0,0 +1,131 @@
+from smartsim.settings import LaunchSettings
+from smartsim.settings.translators.launch.mpi import MpiArgBuilder, MpiexecArgBuilder, OrteArgBuilder
+import pytest
+import itertools
+from smartsim.settings.launchCommand import LauncherType
+
+@pytest.mark.parametrize(
+    "launcher",
+    [
+        pytest.param(LauncherType.Mpirun, id="launcher_str_mpirun"),
+        pytest.param(LauncherType.Mpiexec, id="launcher_str_mpiexec"),
+        pytest.param(LauncherType.Orterun, id="launcher_str_orterun"),
+    ],
+)
+def test_launcher_str(launcher):
+    """Ensure launcher_str returns appropriate value"""
+    ls = LaunchSettings(launcher=launcher)
+    assert ls.launch_args.launcher_str() == launcher.value
+
+@pytest.mark.parametrize(
+    "l,function,value,result,flag",
+    [
+    # Use OpenMPI style settigs for all launchers
+    *itertools.chain.from_iterable(
+        (
+            (
+            pytest.param(l, "set_walltime", ("100",),"100","timeout",id="set_walltime"),
+            pytest.param(l, "set_task_map", ("taskmap",),"taskmap","map-by",id="set_task_map"),
+            pytest.param(l, "set_cpus_per_task", (2,),"2","cpus-per-proc",id="set_cpus_per_task"),
+            pytest.param(l, "set_cpu_binding_type", ("4",),"4","bind-to",id="set_cpu_binding_type"),
+            pytest.param(l, "set_tasks_per_node", (4,),"4","npernode",id="set_tasks_per_node"),
+            pytest.param(l, "set_tasks", (4,),"4","n",id="set_tasks"),
+            pytest.param(l, "set_executable_broadcast", ("broadcast",),"broadcast","preload-binary",id="set_executable_broadcast"),
+            pytest.param(l, "set_hostlist", ("host_A",),"host_A","host",id="set_hostlist_str"),
+            pytest.param(l, "set_hostlist", (["host_A","host_B"],),"host_A,host_B","host",id="set_hostlist_list[str]"),
+            pytest.param(l, "set_hostlist_from_file", ("./path/to/hostfile",),"./path/to/hostfile","hostfile",id="set_hostlist_from_file"),
+        )
+                for l in ([LauncherType.Mpirun, MpiArgBuilder], [LauncherType.Mpiexec, MpiexecArgBuilder], [LauncherType.Orterun, OrteArgBuilder])
+            ))
```

Comment:
Nit: black please! Its hard to match the open/close parentheses with the multiple indent levels